### PR TITLE
Accept Spaces in Message Types

### LINKED ISSUE #64: InvalidCharacterError when passing type.
Clang has a 'critical error' message type. When I pass this to the base linter package it throws an Exception saying it has a critical error. I realize that the space in the type is the issue here, but I was kinda hoping that the base linter package could replace unfriendly characters with friendly characters.

### PR DESCRIPTION
**Pull request by steelbrain**
Fixes #64

### CODE REVIEW

**Comment by iam4x** [comment]
:+1:

File: lib/editor-linter.coffee [hunk 1]
```diff
@@ -79,6 +79,7 @@ class EditorLinter
       unless result.type
         throw new Error "Missing type field on Linter Response, Got: #{Object.keys(result)}"
       result.range = Range.fromObject result.range if result.range?
+      result.class = result.type.toLowerCase().replace(' ', '-')
```
Comment (AsaAyers) [inline lib/editor-linter.coffee]: Have you documented somewhere that the type is lowercased and spaces become dashes?

Comment (steelbrain) [inline lib/editor-linter.coffee]: Yep

Comment (steelbrain) [inline lib/editor-linter.coffee]: Check the Wiki

**Pull request merged**